### PR TITLE
Change SQS event handlers to take eventType from MessageAttributes instead of the body

### PR DESCRIPTION
**Pull request by smcveigh941**
### Background

Domain and prison events work fine in production. These events have the `eventType` inside the body of the SQS message. Probation events **do not**. This means our probation event handler is wrong, and is not properly processing the events raised in production.

See how offender-events-ui gets its event type for all events [here](https://github.com/ministryofjustice/offender-events-ui/blob/7c28797020e5fdd4f500c21aa44829b081f07562/src/main/kotlin/uk/gov/justice/hmpps/offenderevents/service/OffenderEventListener.kt#L32)

---
 
### Fix

- All 3 events have the `eventType` in the `MessageAttributes` of the SQS message, so we should get the message type from there instead.
- Integration tests to verify change

### CODE REVIEW

File: server/listeners/eventHandlers/probationEvents/offenderManagerChangedEventHandler.ts [hunk 1]
```diff
@@ -1,15 +1,11 @@
-import { ProbationEvent } from '../../../@types/events'
+import { ProbationEventMessage } from '../../../@types/events'
 import CommunityService from '../../../services/communityService'
 import LicenceService from '../../../services/licenceService'
 
 export default class OffenderManagerChangedEventHandler {
   constructor(private readonly communityService: CommunityService, private readonly licenceService: LicenceService) {}
 
-  // TODO: I suspect this is an incorrect interpretation of the meaning of this event?
```
Comment (smcveigh941) [inline server/listeners/eventHandlers/probationEvents/offenderManagerChangedEventHandler.ts:8]: Think we have agreed that this listener is doing as it should, and another listener would be added to handle the OFFICER_DETAIL_CHANGED event

File: integration_tests/integration/eventHandlers.spec.ts [hunk 2]
```diff
@@ -50,11 +62,18 @@ context('Event handlers', () => {
     it('should listen to the offender manager changed event and call endpoint to update responsible COM', () => {
       cy.task('stubGetAnOffendersManagers')
       cy.task('stubGetStaffDetailsByStaffId')
+      cy.task('stubUpdateResponsibleCom')
```
Comment (smcveigh941) [inline integration_tests/integration/eventHandlers.spec.ts:65]: This was not already stubbed. The test did not fail for some reason.. but it threw a warning

File: server/listeners/eventHandlers/probationEvents/offenderManagerChangedEventHandler.ts (hunk 1 above)
Comment (timharrison-moj) [reply server/listeners/eventHandlers/probationEvents/offenderManagerChangedEventHandler.ts:8]: Yes, agreed.

File: integration_tests/integration/eventHandlers.spec.ts [hunk 3]
```diff
@@ -13,7 +13,13 @@ context('Event handlers', () => {
       cy.task(
         'sendDomainEvent',
         `{
-            "Message" :  "{\\"eventType\\":\\"prison-offender-events.prisoner.released\\",\\"additionalInformation\\":{\\"nomsNumber\\":\\"A7774DY\\",\\"reason\\":\\"RELEASED\\",\\"details\\":\\"Movement reason code CR\\",\\"currentLocation\\":\\"OUTSIDE_PRISON\\",\\"prisonId\\":\\"MDI\\",\\"currentPrisonStatus\\":\\"NOT_UNDER_PRISON_CARE\\"},\\"version\\":1,\\"occurredAt\\":\\"2022-01-12T14:56:51.662128Z\\",\\"publishedAt\\":\\"2022-01-12T14:58:25.021008001Z\\",\\"description\\":\\"A prisoner has been released from prison\\"}"
+            "Message": "{\\"additionalInformation\\":{\\"nomsNumber\\":\\"A7774DY\\",\\"reason\\":\\"RELEASED\\",\\"details\\":\\"Movement reason code CR\\",\\"currentLocation\\":\\"OUTSIDE_PRISON\\",\\"prisonId\\":\\"MDI\\",\\"currentPrisonStatus\\":\\"NOT_UNDER_PRISON_CARE\\"},\\"version\\":1,\\"occurredAt\\":\\"2022-01-12T14:56:51.662128Z\\",\\"publishedAt\\":\\"2022-01-12T14:58:25.021008001Z\\",\\"description\\":\\"A prisoner has been released from prison\\"}",
```
Comment (timharrison-moj) [inline integration_tests/integration/eventHandlers.spec.ts:16]: I thought domain events were ok and it was the probation events that were the issue?

**Review by timharrison-moj — APPROVED** [review]
LGTM

**Pull request merged**